http-tracer.go

```diff
@@ -248,6 +248,20 @@ func Trace(f http.HandlerFunc, logBody bool, w http.ResponseWriter, r *http.Requ
 	rw.LogBody = logBody
 	f(rw, r)
 
+	if backend.healthOptimistic {
+		// when running in optimistic mode the node will be taken out
+		// of the pool when it returns one of the following HTTP
+		// response status:
+		// - 502 Bad gateway (i.e. can't connect to remote)
+		// - 503 Service unavailable
+		// - 504 Gateway timeout
+		if rw.StatusCode == http.StatusBadGateway || rw.StatusCode == http.StatusServiceUnavailable || rw.StatusCode == http.StatusGatewayTimeout {
+			if backend.setOffline() {
+				go backend.healthCheck()
+			}
+		}
+	}
```

Comment:
__PROBABLY__ there is a better place this check AFAIK, the reverse proxy has an error handler, it may be useful to use that instead of this one

for example, what happens when a node goes offline ? is this use case handled as well ?

Reply:
If a node goes offline, then the sidecar's reverse proxy will actually return 502 ([source](https://github.com/ramondeklein/sidekick/blob/optimistic-mode/reverse/proxy.go#L193)), so that will be handled properly. I agree that `http-tracer.go` doesn't sound like the place to catch this, but in `proxy.go` I don't have a reference to `backend` anymore. That's why I put it here.